### ISSUE TITLE
[macOS, Qt6] WARNING: skipping QEventPoint when using trackpad

### ISSUE THREAD

**Issue by aganders3** [vispy/vispy#2455]
Qt6 is printing these warnings when interacting with a vispy visual on a mac using a touchpad.
```
WARNING: skipping QEventPoint(id=1 ts=0 pos=0,0 scn=1026.88,553.308 gbl=1026.88,553.308 Released ellipse=(1x1 ∡ 0) vel=0,0 press=-1026.88,-553.308 last=-1026.88,-553.308 Δ 1026.88,553.308) : no target window
15:05:15 WARNING skipping QEventPoint(id=1 ts=0 pos=0,0 scn=1026.88,553.308 gbl=1026.88,553.308 Released ellipse=(1x1 ∡ 0) vel=0,0 press=-1026.88,-553.308 last=-1026.88,-553.308 Δ 1026.88,553.308) : no target window
WARNING: skipping QEventPoint(id=1 ts=0 pos=0,0 scn=1152.25,525.882 gbl=1152.25,525.882 Released ellipse=(1x1 ∡ 0) vel=0,0 press=-1152.25,-525.882 last=-1152.25,-525.882 Δ 1152.25,525.882) : no target window
15:05:17 WARNING skipping QEventPoint(id=1 ts=0 pos=0,0 scn=1152.25,525.882 gbl=1152.25,525.882 Released ellipse=(1x1 ∡ 0) vel=0,0 press=-1152.25,-525.882 last=-1152.25,-525.882 Δ 1152.25,525.882) : no target window
WARNING: skipping QEventPoint(id=1 ts=0 pos=0,0 scn=1302.54,597.871 gbl=1302.54,597.871 Released ellipse=(1x1 ∡ 0) vel=0,0 press=-1302.54,-597.871 last=-1302.54,-597.871 Δ 1302.54,597.871) : no target window
```

From what I can tell you can reproduce this by performing a two-finger gesture (e.g. pinch-to-zoom) _outside_ the vispy area (but with the window in focus), then moving the cursor back over it.

[There is a related topic on the Qt bug tracker](https://bugreports.qt.io/browse/QTBUG-103935), but it's unclear to me if it's being treated as a bug. Their workaround (turn off `WA_AcceptTouchEvents`) is effective, but obviously removes the possibility to support two-finger gestures like pinch-to-zoom or rotate. There is also suggestion to use `QNativeGestureEvent` instead, but I can't tell which to prefer as the documentation is quite thin and searching reveals [answers like this one](https://stackoverflow.com/a/40289162/333308) (to my eyes it's arguable which is the "higher level" API).

I was able to mostly re-implement the events with `QNativeGestureEvent` and the warnings indeed go away, but it lacks some of the nice features of [`QPinchGesture`](https://doc.qt.io/qt-6/qpinchgesture.html) such as `lastScaleFactor`, `totalScaleFactor`, and `totalRotationAngle`. I can open a draft PR for this if there is interest.

As far as I can tell these touch events are not hooked up to anything in vispy, and without #1980 I'm not sure they can be hooked up in other applications either. Still, it would be nice to silence the warnings and to use the proper Qt API if it can be determined. Without #1980 I'm not clear if changing this would affect any downstream users.

See https://github.com/napari/napari/issues/5530 for an initial report of this issue.

There are also related vispy issues:
https://github.com/vispy/vispy/issues/721
https://github.com/vispy/vispy/issues/2012

**Comment by djhoese** [comment]
Hm :thinking: this is difficult to know what is the best solution without doing a deep dive into Qt expected usage. It sounds like you've done quite of that work already. In that last SO answer you linked to it isn't clear to me what the difference between `NativeGesture` (what the SO answer talks about using) and `Gesture` (what the Qt docs they link to use) is.

If you have modifications that could fix this without breaking anything for other platforms then I'm definitely curious to see it (in a PR). I don't have a macbook and don't use gesture handling in any of my projects so I'm not really familiar with common ways to solve these types of problems. Whatever you, the napari team, or anyone else can get working will likely be OK with me as long as it is "mostly" backwards compatible.

@brisvag and @kmuehlbauer let me know if you have an opinion on this.

@almarklein do you have a mac?

**Comment by psobolewskiPhD** [comment]
I think this post in the QT bug thread is more instructive than the SO:
https://bugreports.qt.io/browse/QTBUG-103935?focusedCommentId=670393&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-670393

As I understand it, touchpad/trackpad events shouldn't be handled as touch events, those are meant to work with touch screens. So `WA_AcceptTouchEvents` is supposed to be used for the later, while native gestures (those passed from the OS) for the former. 
Does anyone with a touch screen device to check whether the current gesture stuff works there? Ironically, I think the touch stuff right now is behind a `darwin` check and there are no touch screen macOS devices...

**Comment by aganders3** [comment]
Thanks @psobolewskiPhD - I agree that post is more instructive. I included the SO just because of the conviction in the _other_ direction with zero justification.

> Oh, could/should we just ignore the warning (assuming we can tell Qt to do that)?

I figured we couldn't do this but it's a reasonable idea so I looked into it and I think we can do it here if we decide that's the way to go:
https://github.com/vispy/vispy/blob/c0d7457a7b1717d47baa8b8622faa2c67bbe4be0/vispy/app/backends/_qt.py#L208

**Comment by djhoese** [comment]
I think I'm OK with either approach as long as everything actually works as expected. Do we want mouse/trackpad and touch gestures to work in vispy (and napari)? I only skimmed that Qt page so not sure I understood as much of this as needed to make a real suggestion or have a real opinion.

If the warning suggests we have something misconfigured, let's re-configure. If the warning suggests something that Qt has trouble handling, then let's ignore it. If some user interaction doesn't work that we want to work, let's fix it.

**Comment by djhoese** [comment]
Ah just saw your PR @aganders3. I'll take a look...

**Comment by aganders3** [comment]
Closing as fixed by #2456 and #2483.
